compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -306,11 +306,11 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                 bx.bitcast(imm, to_backend_ty)
             }
             (Pointer(..), Pointer(..)) => bx.pointercast(imm, to_backend_ty),
-            (Int(..), Pointer(..)) => bx.inttoptr(imm, to_backend_ty),
+            (Int(..), Pointer(..)) => bx.gep(bx.type_i8(), bx.const_null(bx.type_ptr()), &[imm]),
```

Comment:
```suggestion
            (Int(..), Pointer(..)) => bx.ptradd(bx.const_null(bx.type_ptr()), &[imm]),
```
Should be able to use this recently added helper.